crates/bevy_render/src/render_phase/draw_state.rs

```diff
@@ -340,12 +340,28 @@ impl<'a> TrackedRenderPass<'a> {
     /// Set the rendering viewport to the given [`Camera`](crate::camera::Viewport) [`Viewport`].
     ///
     /// Subsequent draw calls will be projected into that viewport.
-    pub fn set_camera_viewport(&mut self, viewport: &Viewport) {
+    pub fn set_camera_viewport(
+        &mut self,
+        viewport: &Viewport,
+        physical_target_size: Option<bevy_math::UVec2>,
+    ) {
+        let physical_size = viewport
+            .physical_size
+            .try_as_absolute(physical_target_size)
+            .expect("Couldn't get camera.physical_target_size (needed for relative viewport size)");
```

Comment:
Hmm, I feel like if this method takes an Option<> then it should handle it gracefully, not panic. So I guess either we need to find a way to handle this gracefully, or this shouldn’t take an Option.

Reply:
First of all, I'm not sure when `physical_target_size` can be `None`. 
I think this is the only place where it's set: 
https://github.com/bevyengine/bevy/blob/9c116d557df5a88940ed2e29dab6ff13c84d61a9/crates/bevy_render/src/camera/camera.rs#L424
So it looks like it's always `Some`.
Maybe it shouldn't be an `Option` at all?

Also, `try_as_absolute` takes an `Option` because this value isn't required if the size/position is an absolute value. So requiring this to always be some value isn't great either.  (Because absolute value without context is still valid)

Reply:
It does appear that it's always `Some` as far as I can tell, since we wouldn't even extract the window if it's not. I think this can be refactored on `ExtractedView`.